exonum-java-binding/testkit/src/main/java/com.exonum.binding.testkit/TestKit.java

```diff
@@ -0,0 +1,197 @@
+  }
+
+  /**
+   * Creates a TestKit network with a single validator node for a single service.
+   */
+  public static TestKit forService(Class<? extends ServiceModule> serviceModule) {
+    return new TestKit(Collections.singletonList(serviceModule), EmulatedNodeType.VALIDATOR,
+        (short) 0, null);
+  }
+
+  @SuppressWarnings("unchecked")
+  public <T extends Service> T getService(short serviceId, Class<T> serviceClass) {
+    Service service = services.get(serviceId);
+    checkArgument(service.getClass().equals(serviceClass), "Service with id %s is not of expected class %s",
+        serviceId, serviceClass.getCanonicalName());
+    return (T) services.get(serviceId);
+  }
+
+  /**
+   * Creates a service from the service module.
+   *
+   * @param moduleClass a class of the user service module
+   */
+  private static UserServiceAdapter createUserModule(Class<? extends ServiceModule> moduleClass) {
+    try {
+      Constructor constructor = moduleClass.getDeclaredConstructor();
+      Object moduleObject = constructor.newInstance();
+      checkArgument(moduleObject instanceof Module, "%s is not a sub-class of %s",
+          moduleClass, Module.class.getCanonicalName());
+      Injector serviceInjector = frameworkInjector.createChildInjector((Module) moduleObject);
+      return serviceInjector.getInstance(UserServiceAdapter.class);
+    } catch (IllegalAccessException e) {
+      throw new IllegalArgumentException("Cannot access the no-arg module constructor", e);
+    } catch (NoSuchMethodException e) {
+      throw new IllegalArgumentException("No no-arg constructor", e);
+    } catch (InstantiationException | InvocationTargetException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private native long nativeCreateTestKit(UserServiceAdapter[] services, boolean auditor,
+      short withValidatorCount, TimeProvider timeProvider);
+
+  private native long nativeCreateSnapshot(long nativeHandle);
+
+  private native Block nativeCreateBlock(long nativeHandle);
+
+  private native Block nativeCreateBlockWithTransactions(long nativeHandle, byte[][] transactions);
+
+  private native EmulatedNode nativeGetEmulatedNode(long nativeHandle);
+
+  /**
+   * Creates a new builder for the TestKit.
+   *
+   * @param nodeType type of the main TestKit node - either validator or auditor. Note that
+   * {@link Service#afterCommit(BlockCommittedEvent)} logic will only be called on the main TestKit node of this
+   * type
+   */
+  public static Builder builder(EmulatedNodeType nodeType) {
+    checkNotNull(nodeType);
+    return new Builder(nodeType);
+  }
+
+  /**
+   * Builder for the TestKit.
+   */
+  public static final class Builder {
+
+    private EmulatedNodeType nodeType;
+    private short validatorCount;
+    private List<Class<? extends ServiceModule>> services = new ArrayList<>();
+    private TimeProvider timeProvider;
+
+    private Builder(EmulatedNodeType nodeType) {
+      this.nodeType = nodeType;
```

Comment:
In core if TestKit is created with an auditor node and `validator_count` is set to `None` then it's unwrapped to `1`, so there is no way to create a network without at least one validator node. But I agree that it's better to explicitly state the number here.
I updated the code so that if the main node is an auditor, then set `validatorCount` to one, as if the main node is a validator, then `validatorCount` stays at 0. We could rename `validatorCount` to `additionalValidatorCount` in case it's not clear that the main node is not counted here